game_of_life-java/src/main/java/GameOfLife/Canvas.java

```diff
@@ -6,38 +6,38 @@
 class Canvas {
     private final int x;
     private final int y;
-    private Cell[][] canvas;
-    private Cell[][] tempCanvas;
+    private boolean[][] canvas;
+    private boolean[][] tempCanvas;
 
     Canvas(int x, int y) {
         this.x = x;
         this.y = y;
         // set two canvasses up, such that one can be calculated from,
         // and the resulting state can be stored in the other
-        this.canvas = new Cell[x][y];
-        this.tempCanvas = new Cell[x][y];
+        this.canvas = new boolean[x][y];
+        this.tempCanvas = new boolean[x][y];
         // seed canvas with randomness
         Random random = new Random(new Date().getTime());
         for (int i = 0; i < x; i++) {
             for (int j = 0; j < y; j++) {
                 // 50/50 chance of each cell being alive at initial configuration
-                this.canvas[i][j] = new Cell(random.nextInt(2) == 0);
+                this.canvas[i][j] = (random.nextInt(2) == 0);
                 // initialise tempCanvas to false, doesn't affect runtime, but each cell needs to have a Cell
-                this.tempCanvas[i][j] = new Cell(false);
+                this.tempCanvas[i][j] = false;
             }
         }
     }
 
     void tick() {
         int neighbours;
         // temp canvas-array for swapping before return
-        Cell[][] temp;
+        boolean[][] temp;
         for (int i = 0; i < this.x; i++) {
             for (int j = 0; j < this.y; j++) {
                 neighbours = countMooreNeighbours(i, j);
 
                 // set new state in tempCanvas based on the life of the given cell in canvas
-                tempCanvas[i][j].setState(cellLife(canvas[i][j], neighbours));
+                tempCanvas[i][j]=(cellLife(canvas[i][j], neighbours));
```

Comment:
missing space around assignment operator

Reply:
Fixed in d984780